pages/instructorQuestionStatistics/instructorQuestionStatistics.js

```diff
@@ -1,138 +1,106 @@
+
+const router = express.Router();
 const sql = sqldb.loadSqlEquiv(__filename);
 
-const filenames = function (locals) {
+const setFilenames = function (locals) {
   const prefix = sanitizeName.questionFilenamePrefix(locals.question, locals.course);
-  return {
-    questionStatsCsvFilename: prefix + 'stats.csv',
-  };
+  locals.questionStatsCsvFilename = prefix + 'stats.csv';
 };
 
-router.get('/', function (req, res, next) {
-  async.series(
-    [
-      (callback) => {
-        debug('set filenames');
-        _.assign(res.locals, filenames(res.locals));
-        callback(null);
-      },
-      (callback) => {
-        sqldb.query(
-          sql.assessment_question_stats,
-          { question_id: res.locals.question.id },
-          function (err, result) {
-            if (ERR(err, callback)) return;
-            res.locals.assessment_stats = result.rows;
-            callback(null);
-          }
-        );
-      },
-      (callback) => {
-        res.locals.question_attempts_histogram = null;
-        res.locals.question_attempts_before_giving_up_histogram = null;
-        res.locals.question_attempts_histogram_hw = null;
-        res.locals.question_attempts_before_giving_up_histogram_hw = null;
-        // res.locals.question_attempts_histogram = res.locals.result.question_attempts_histogram;
-        // res.locals.question_attempts_before_giving_up_histogram = res.locals.result.question_attempts_before_giving_up_histogram;
-        // res.locals.question_attempts_histogram_hw = res.locals.result.question_attempts_histogram_hw;
-        // res.locals.question_attempts_before_giving_up_histogram_hw = res.locals.result.question_attempts_before_giving_up_histogram_hw;
-        callback(null);
-      },
-    ],
-    (err) => {
-      if (ERR(err, next)) return;
-      res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-    }
-  );
-});
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    setFilenames(res.locals);
+    const statsResult = await sqldb.queryAsync(sql.assessment_question_stats, {
+      question_id: res.locals.question.id,
+    });
+    res.locals.assessment_stats = statsResult.rows;
 
-router.get('/:filename', function (req, res, next) {
-  _.assign(res.locals, filenames(res.locals));
+    res.locals.question_attempts_histogram = null;
+    res.locals.question_attempts_before_giving_up_histogram = null;
+    res.locals.question_attempts_histogram_hw = null;
+    res.locals.question_attempts_before_giving_up_histogram_hw = null;
+    // res.locals.question_attempts_histogram = res.locals.result.question_attempts_histogram;
+    // res.locals.question_attempts_before_giving_up_histogram = res.locals.result.question_attempts_before_giving_up_histogram;
+    // res.locals.question_attempts_histogram_hw = res.locals.result.question_attempts_histogram_hw;
+    // res.locals.question_attempts_before_giving_up_histogram_hw = res.locals.result.question_attempts_before_giving_up_histogram_hw;
```

Comment:
👀 any idea what's up with this?

Reply:
I guess this is some leftover thing from the student who originally implemented this? I guess it was never implemented, although they do look like interesting things to be computing.

Reply:
Gotcha - I'll rip this out in another PR!